componentLibraries/defaultLibrary/Special/AeroComponents/AeroFlightRecorder.hpp

```diff
@@ -0,0 +1,329 @@
+    limitations under the License.
+
+
+ The full license is available in the file LICENSE.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   AeroFlightRecorder.hpp
+//! @author Alejandro Sobron <alejandro.sobron@liu.se>
+//! @date   2020-05-15
+//!
+//! @brief Contains a log-file generator component for geographical visualization
+//! @ingroup AeroComponents
+//!
+
+#ifndef AEROFLIGHTRECORDER_HPP_INCLUDED
+#define AEROFLIGHTRECORDER_HPP_INCLUDED
+
+#include "ComponentEssentials.h"
+#include "ComponentSystem.h"
+#include "ComponentUtilities.h"
+#include "math.h"
+#include <iostream>
+#include <fstream>
+#include <ctime>
+
+namespace hopsan {
+
+    class AeroFlightRecorder : public ComponentSignal
+    {
+
+    private:
+    // Private member variables
+
+        //inputVariables pointers
+        double *mpLongitude;
+        double *mpLatitude;
+        double *mpAltitude;
+        double *mpPhi;
+        double *mpTheta;
+        double *mpPsi;
+        double *mpV;
+
+        //inputParameters pointers
+        double *mpDt;
+        double *mpScaleX;
+        double *mpScaleY;
+        double *mpScaleZ;
+
+        //inputVariables
+        double longitude;
+        double latitude;
+        double altitude;
+        double roll;
+        double pitch;
+        double heading;
+        double speed;
+
+        //inputParameters
+        double dt;
+        double scaleX;
+        double scaleY;
+        double scaleZ;
+
+        // Constants
+        enum FileTypes {FlightPathKML, GroundPathKML, ThreeDmodelKML};
+        int mFileType;
+        HString mFilePath;
+        HString mDaeModelPath;
+        std::ofstream mFile;
+
+        // Other operating variables
+        double tilt;
+        double mLastLogTime;
+
+    public:
+    // The creator function that is registered when a component lib is loaded into Hopsan
+        static Component *Creator()
+        {
+            return new AeroFlightRecorder();
+        }
+
+        void configure()
+        {
+            //Add ports to the component
+
+            //Add inputVariables to the component
+            addInputVariable("longitude","Longitude E","deg",15.577,&mpLongitude);
+            addInputVariable("latitude","Latitude N","deg",58.401,&mpLatitude);
+            addInputVariable("altitude","Altitude","m",0,&mpAltitude);
+            addInputVariable("phi","Roll","rad",0,&mpPhi);
+            addInputVariable("theta","Pitch","rad",0,&mpTheta);
+            addInputVariable("psi","Heading","rad",0,&mpPsi);
+            addInputVariable("v","Speed","m/s",0,&mpV);
+
+            // Add inputParameters
+            addInputVariable("dt", "Logging interval (0 = log every sample)", "s", 1, &mpDt);
```

Comment:
Instead of s for seconds use Quantity Time
Scaling below, is scale really m, should it not be without dimension, (leave blank)

Reply:
Absolutely. You right with the "scale factors". I just wasn't sure I could leave it blank and "length" was the closest concept ;)
I'll fix this right away.